### ISSUE TITLE
Update Steps Calculation When Using Gradient Accumulation

### ISSUE THREAD

**Issue by whikwon** [Vchitect/Latte#134]
Should the calculation for the update steps per epoch be adjusted to account for gradient accumulation? Specifically, wouldn’t it be more accurate to use the formula:
```
num_update_steps_per_epoch = math.ceil(len(loader)) / args.gradient_accumulation_steps
```

Code here: https://github.com/Vchitect/Latte/blob/07c7e5c64d25f627766fa69bea50f3a27900ec84/train.py#L175-L177

**Comment by maxin-cn** [comment]
This current version of the Latte training code does not fully support gradient accumulation. You can modify the training code of Latte by referring to [the training script](https://github.com/huggingface/diffusers/blob/1a04812439c82a9dd318d14a800bb04e84dbbfc0/examples/text_to_image/train_text_to_image.py#L828) provided by diffusers.

**Comment by whikwon** [comment]
Thank you for your prompt reply! I’ll review the reference you shared.
